### PR TITLE
Adding in service to return AZ listing for a region with tests.

### PR DESCRIPTION
**Pull request by andrewpowell**
Adds the service to return the list of AWS Availability Zones for a region.  Returns 404 on an invalid region in the region config.

### CODE REVIEW

File: src/main/java/com/atlassian/migration/datacenter/core/aws/region/AvailabilityZoneManager.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import java.util.List;
+
+@Service
+public class AvailabilityZoneManager implements AvailabilityZoneService {
+
+    private final AtlassianPluginAWSCredentialsProvider credentialsProvider;
+    private final GlobalInfrastructure globalInfrastructure;
+
+    @Autowired
```
Comment (nanux) [inline src/main/java/com/atlassian/migration/datacenter/core/aws/region/AvailabilityZoneManager.java]: I don't think this is necessary as it is defined in class with `@Service` annotation.

File: src/main/java/com/atlassian/migration/datacenter/core/aws/region/AvailabilityZoneManager.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.ec2.Ec2Client;
+import software.amazon.awssdk.services.ec2.model.AvailabilityZone;
+import software.amazon.awssdk.services.ec2.model.DescribeAvailabilityZonesResponse;
+
+import java.util.List;
+
+@Service
```
Comment (nanux) [inline src/main/java/com/atlassian/migration/datacenter/core/aws/region/AvailabilityZoneManager.java]: /nit We are using `@Component` for the other beans. I think using `Service` might make sense but I would prefer consistency and then have a sweeping change to change all Services to `@Service`. It is only semantic anyway.

File: src/main/java/com/atlassian/migration/datacenter/api/aws/AWSAvailabilityZoneEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        Region searchRegion = Region.of(this.regionService.getRegion());
+        return this.findAZListForRegion(searchRegion);
+    }
+
+    private Response findAZListForRegion(Region region) {
+        try {
+            List<AvailabilityZone> zonesList = this.availabilityZoneService.getAZForRegion(region);
+            try (Stream<AvailabilityZone> azStream = zonesList.parallelStream()) {
+                List<String> nameList = azStream.map(AvailabilityZone::zoneName).sorted().collect(Collectors.toList());
+                return Response.status(Response.Status.OK).entity(nameList).build();
```
Comment (nanux) [inline src/main/java/com/atlassian/migration/datacenter/api/aws/AWSAvailabilityZoneEndpoint.java]: can be shorten to `Response.ok(nameList).build`

File: src/main/java/com/atlassian/migration/datacenter/core/aws/region/AvailabilityZoneManager.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+    }
+
+    private Ec2Client getEC2Client(Region region) {
+        return Ec2Client.builder().region(region).credentialsProvider(this.credentialsProvider).build();
+    }
+
+    private boolean isValidRegion(String testRegion) {
+        return this.globalInfrastructure.
```
Comment (nanux) [inline src/main/java/com/atlassian/migration/datacenter/core/aws/region/AvailabilityZoneManager.java]: /nit
```
this.globalInfrastructure
                .getRegions()
```
the dot should be with the `getRegions` method.

File: src/main/java/com/atlassian/migration/datacenter/api/aws/AWSAvailabilityZoneEndpoint.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getAvailabilityZoneList(@PathParam("region") String region) {
+        Region searchRegion = Region.of(this.regionService.getRegion());
+        return this.findAZListForRegion(searchRegion);
+    }
+
+    private Response findAZListForRegion(Region region) {
+        try {
+            List<AvailabilityZone> zonesList = this.availabilityZoneService.getAZForRegion(region);
+            try (Stream<AvailabilityZone> azStream = zonesList.parallelStream()) {
```
[the remark this answers is not in the document]
Comment (andrewpowell) [reply src/main/java/com/atlassian/migration/datacenter/api/aws/AWSAvailabilityZoneEndpoint.java]: fixed

File: src/main/java/com/atlassian/migration/datacenter/core/aws/region/AvailabilityZoneManager.java (hunk 2 above)
Comment (andrewpowell) [reply src/main/java/com/atlassian/migration/datacenter/core/aws/region/AvailabilityZoneManager.java]: fixed

File: src/main/java/com/atlassian/migration/datacenter/api/aws/AWSAvailabilityZoneEndpoint.java (hunk 3 above)
Comment (andrewpowell) [reply src/main/java/com/atlassian/migration/datacenter/api/aws/AWSAvailabilityZoneEndpoint.java]: fixed

File: src/main/java/com/atlassian/migration/datacenter/core/aws/region/AvailabilityZoneManager.java (hunk 4 above)
Comment (andrewpowell) [reply src/main/java/com/atlassian/migration/datacenter/core/aws/region/AvailabilityZoneManager.java]: fixed

**Comment by theghostwhoforks** [comment]
Hey @andrewpowell - @nanux just merged this PR; going forward, do feel free to merge the PR once the necessary approvals are there and the builds are green!

**Pull request merged**